[2024-08-31T18:24:47.191+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_retail_analysis_dag.spark_retail_analysis_task manual__2024-08-31T18:24:41.188518+00:00 [queued]>
[2024-08-31T18:24:47.217+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_retail_analysis_dag.spark_retail_analysis_task manual__2024-08-31T18:24:41.188518+00:00 [queued]>
[2024-08-31T18:24:47.218+0000] {taskinstance.py:2193} INFO - Starting attempt 1 of 1
[2024-08-31T18:24:47.274+0000] {taskinstance.py:2214} INFO - Executing <Task(SparkSubmitOperator): spark_retail_analysis_task> on 2024-08-31 18:24:41.188518+00:00
[2024-08-31T18:24:47.283+0000] {standard_task_runner.py:60} INFO - Started process 15447 to run task
[2024-08-31T18:24:47.356+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'spark_retail_analysis_dag', 'spark_retail_analysis_task', 'manual__2024-08-31T18:24:41.188518+00:00', '--job-id', '53', '--raw', '--subdir', 'DAGS_FOLDER/spark_retail_analysis_dag.py', '--cfg-path', '/tmp/tmpqh66uwx5']
[2024-08-31T18:24:47.369+0000] {standard_task_runner.py:88} INFO - Job 53: Subtask spark_retail_analysis_task
[2024-08-31T18:24:47.490+0000] {logging_mixin.py:188} WARNING - /home/airflow/.local/lib/python3.9/site-packages/airflow/settings.py:194 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-08-31T18:24:47.709+0000] {task_command.py:423} INFO - Running <TaskInstance: spark_retail_analysis_dag.spark_retail_analysis_task manual__2024-08-31T18:24:41.188518+00:00 [running]> on host dataeng-airflow-scheduler
[2024-08-31T18:24:48.767+0000] {taskinstance.py:2510} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='dibimbing' AIRFLOW_CTX_DAG_ID='spark_retail_analysis_dag' AIRFLOW_CTX_TASK_ID='spark_retail_analysis_task' AIRFLOW_CTX_EXECUTION_DATE='2024-08-31T18:24:41.188518+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-08-31T18:24:41.188518+00:00'
[2024-08-31T18:24:48.837+0000] {base.py:83} INFO - Using connection ID 'spark_main' for task execution.
[2024-08-31T18:24:48.841+0000] {spark_submit.py:403} INFO - Spark-Submit cmd: spark-submit --master spark://dataeng-spark-master:7077 --packages org.postgresql:postgresql:42.2.18 --name arrow-spark /spark-scripts/spark-retail-analysis.py
[2024-08-31T18:25:14.781+0000] {spark_submit.py:579} INFO - :: loading settings :: url = jar:file:/home/airflow/.local/lib/python3.9/site-packages/pyspark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-08-31T18:25:15.722+0000] {spark_submit.py:579} INFO - Ivy Default Cache set to: /home/airflow/.ivy2/cache
[2024-08-31T18:25:15.735+0000] {spark_submit.py:579} INFO - The jars for the packages stored in: /home/airflow/.ivy2/jars
[2024-08-31T18:25:15.754+0000] {spark_submit.py:579} INFO - org.postgresql#postgresql added as a dependency
[2024-08-31T18:25:15.770+0000] {spark_submit.py:579} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-ff5cd7f4-188e-4dc4-8f6e-f81a303d4bb0;1.0
[2024-08-31T18:25:15.787+0000] {spark_submit.py:579} INFO - confs: [default]
[2024-08-31T18:25:16.530+0000] {spark_submit.py:579} INFO - found org.postgresql#postgresql;42.2.18 in central
[2024-08-31T18:25:16.873+0000] {spark_submit.py:579} INFO - found org.checkerframework#checker-qual;3.5.0 in central
[2024-08-31T18:25:17.306+0000] {spark_submit.py:579} INFO - :: resolution report :: resolve 1422ms :: artifacts dl 114ms
[2024-08-31T18:25:17.318+0000] {spark_submit.py:579} INFO - :: modules in use:
[2024-08-31T18:25:17.319+0000] {spark_submit.py:579} INFO - org.checkerframework#checker-qual;3.5.0 from central in [default]
[2024-08-31T18:25:17.320+0000] {spark_submit.py:579} INFO - org.postgresql#postgresql;42.2.18 from central in [default]
[2024-08-31T18:25:17.320+0000] {spark_submit.py:579} INFO - ---------------------------------------------------------------------
[2024-08-31T18:25:17.321+0000] {spark_submit.py:579} INFO - |                  |            modules            ||   artifacts   |
[2024-08-31T18:25:17.445+0000] {spark_submit.py:579} INFO - |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2024-08-31T18:25:17.464+0000] {spark_submit.py:579} INFO - ---------------------------------------------------------------------
[2024-08-31T18:25:17.480+0000] {spark_submit.py:579} INFO - |      default     |   2   |   0   |   0   |   0   ||   2   |   0   |
[2024-08-31T18:25:17.496+0000] {spark_submit.py:579} INFO - ---------------------------------------------------------------------
[2024-08-31T18:25:17.504+0000] {spark_submit.py:579} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-ff5cd7f4-188e-4dc4-8f6e-f81a303d4bb0
[2024-08-31T18:25:17.513+0000] {spark_submit.py:579} INFO - confs: [default]
[2024-08-31T18:25:17.517+0000] {spark_submit.py:579} INFO - 0 artifacts copied, 2 already retrieved (0kB/25ms)
[2024-08-31T18:25:19.915+0000] {spark_submit.py:579} INFO - 24/08/31 18:25:19 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-08-31T18:25:21.121+0000] {spark_submit.py:579} INFO - File "/spark-scripts/spark-retail-analysis.py", line 71
[2024-08-31T18:25:21.122+0000] {spark_submit.py:579} INFO - 
[2024-08-31T18:25:21.124+0000] {spark_submit.py:579} INFO - ^
[2024-08-31T18:25:21.130+0000] {spark_submit.py:579} INFO - SyntaxError: unexpected EOF while parsing
[2024-08-31T18:25:21.140+0000] {spark_submit.py:579} INFO - 24/08/31 18:25:21 INFO ShutdownHookManager: Shutdown hook called
[2024-08-31T18:25:21.145+0000] {spark_submit.py:579} INFO - 24/08/31 18:25:21 INFO ShutdownHookManager: Deleting directory /tmp/spark-5406e72b-aba4-40d3-b606-7e88887d8f28
[2024-08-31T18:25:21.386+0000] {taskinstance.py:2728} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://dataeng-spark-master:7077 --packages org.postgresql:postgresql:42.2.18 --name arrow-spark /spark-scripts/spark-retail-analysis.py. Error code is: 1.
[2024-08-31T18:25:21.419+0000] {taskinstance.py:1149} INFO - Marking task as FAILED. dag_id=spark_retail_analysis_dag, task_id=spark_retail_analysis_task, execution_date=20240831T182441, start_date=20240831T182447, end_date=20240831T182521
[2024-08-31T18:25:21.503+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 53 for task spark_retail_analysis_task (Cannot execute: spark-submit --master spark://dataeng-spark-master:7077 --packages org.postgresql:postgresql:42.2.18 --name arrow-spark /spark-scripts/spark-retail-analysis.py. Error code is: 1.; 15447)
[2024-08-31T18:25:21.580+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-08-31T18:25:21.663+0000] {taskinstance.py:3309} INFO - 0 downstream tasks scheduled from follow-on schedule check
